src/httpclient.c

```diff
@@ -316,10 +329,31 @@ int32_t httpclient_getConnectTimeout(void)
 {
     int32_t timeout = DEFAULT_CONNECT_TIMEOUT;
 
-    void *data = corto_threadTlsGet(TLS_ConnectTimeoutKey);
-    if (data) {
-        timeout = (int32_t)(intptr_t)data;
+    httpclient_Config config = (httpclient_Config)corto_threadTlsGet(
+        HTTPCLIENT_KEY_CONFIG);
+    if (config) {
+        timeout = config->connectTimeout;
     }
 
     return timeout;
 }
+
+static void httpclient_tlsConfigFree(void *o) {
+    httpclient_Config config = (httpclient_Config)o;
+    if (config) {
+        free(config);
+    }
+}
+
+int16_t httpclient_construct(void)
```

Comment:
Why is this a separate function? Could this code be moved to the `httpclientMain` function?